x/tx/aminojson/any.go

```diff
@@ -0,0 +1,60 @@
+package aminojson
+
+import (
+	"fmt"
+	"io"
+
+	"github.com/pkg/errors"
+	"google.golang.org/protobuf/proto"
+	"google.golang.org/protobuf/reflect/protoreflect"
+	"google.golang.org/protobuf/reflect/protoregistry"
+)
+
+const (
+	typeUrlName protoreflect.Name = "type_url"
+	valueName   protoreflect.Name = "value"
+)
+
+func (aj AminoJSON) marshalAny(message protoreflect.Message, writer io.Writer) error {
+	fields := message.Descriptor().Fields()
+	typeUrlField := fields.ByName(typeUrlName)
+	if typeUrlField == nil {
+		return fmt.Errorf("expected type_url field")
+	}
+
+	typeUrl := message.Get(typeUrlField).String()
+	resolver := protoregistry.GlobalTypes
+
+	typ, err := resolver.FindMessageByURL(typeUrl)
+	if err != nil {
+		return errors.Wrapf(err, "can't resolve type URL %s", typeUrl)
+	}
+
+	valueField := fields.ByName(valueName)
+	if valueField == nil {
+		return fmt.Errorf("expected value field")
+	}
+
+	valueBz := message.Get(valueField).Bytes()
+
+	valueMsg := typ.New()
+	err = proto.Unmarshal(valueBz, valueMsg.Interface())
+	if err != nil {
+		return err
+	}
+
+	aminoName, named := getMessageName(valueMsg)
+	if !named {
+		return fmt.Errorf("any fields require an amino.name annotation")
```

Comment:
I don't think this is true. You can pack something in an Any without an `amino.name` (or in go-amino lingua, without `RegisterConcrete`). It happened to us on the SDK on some Msgs.

However, it's heavily recommended to add the `amino.name` annotation though. For one, it'll make the message work with CosmJS. So I actually think it's a good idea to error here 👍 

I would change it to:
```suggestion
		return fmt.Errorf("message %s is packed into an any field, so requires an amino.name annotation")
```

(i.e. the message needs amino.name, not the field)